src/bin/common/solving_loop.ml

```diff
@@ -650,13 +714,58 @@ let main () =
       unsupported_opt name
   in
 
+  (* Fetches the term value in the current model. *)
+  let evaluate_term get_value name term =
+    let ae_form =
+      D_cnf.make_form
+        name
+        term
+        Loc.dummy
+        ~decl_kind:Expr.Dgoal
+    in
+    match get_value ae_form with
+    | None -> "unknown" (* Not in the standard, but useful for recording when
+                           Alt-Ergo fails to guess the value of a term. *)
+    | Some v -> Fmt.to_to_string Expr.print v
+  in
+
+  let print_terms_assignments =
+    Fmt.list
+      ~sep:(fun fmt _ -> Fmt.pf fmt "@,")
+      (fun fmt (name, v) -> Fmt.pf fmt "(%s %s)" name v)
+  in
+
+  let handle_get_assignment ~get_value st =
+    let assignments =
+      Util.MS.fold
```

Comment:
Nit: you can probably leverage `Fmt.iter_bindings` here.

Reply:
I preferred to build the list and print it afterwards to make the code more readable